### DISCUSSION TITLE
Status 405 with IDP initiated SSO

### DISCUSSION THREAD

**Discussion by ErwanSC** [keycloak/keycloak#15097] [Q&A]
Hello,

We are trying to set up an IDP initiated SSO with Azure and Keycloak but we get a "405 Method Not Allowed" response with a beautiful "An internal server error has occurred" from Keycloak.
However there is nothing in KC's logs...

Server Version | 18.0.0

This is the browser error :
![image](https://user-images.githubusercontent.com/51318309/197343954-aa61cb6c-5333-4bd9-bf8a-54d4fc3efe65.png)

And KC error : 
![image](https://user-images.githubusercontent.com/51318309/197343972-25685c71-667c-462e-98c1-9fedef0b2f25.png)

Our Azure App configuration : 
![image](https://user-images.githubusercontent.com/51318309/197344047-a79872b1-15ca-4d35-9a25-5518d6ec41c1.png)

The url ending with https://auth.domain.com/auth/realms/realmname/protocol/saml/clients/idpinitiated is OK and we could connect to our application with it. So KC configuration seems to be fine...

![image](https://user-images.githubusercontent.com/51318309/197344399-bd6b1966-4997-4c96-bc5a-333c6927d3e0.png)
![image](https://user-images.githubusercontent.com/51318309/197344440-f8fff0a2-6d0e-4c9b-9b0e-9f419e5e2c29.png)

Why Keycloak logs don't show some errors ?
What is wrong with our configuration ?

Thanks :)

**Comment by ErwanSC** [comment]
Hello, we upgrade ou keycloak from 18.0 to 19.0.2 and problem is still the same...

**Comment by ErwanSC** [comment]
Hello,

After upgrading our Keycloak in 20.0.0, we still have the problem.
Any help please ?

**Reply by TechColumn** [reply]
Any update on this? We are facing the same issue.

**Comment by TechColumn** [comment]
Hi,
Any solution to this problem? Appreciate any help. Thanks

**Comment by ErwanSC** [comment]
Hi,
Sorry, no solution found yet for this problem. We let him down temporarily.

**Comment by renaudpawlak** [comment]
I have a very similar issue (same 405 error with internal error and no logs in KC logs), while trying to use SalesForce as an external IDP. Everything looks ok on SalesForce side until going back to KC... Anyone made it work?

**Reply by Jeff-Tian** [reply]
I'm interested in this issue. Could you provide a screenshot of the browser's developer tool highlighting the **request headers** and **response headers**?

It seems you were meeting the issue that the callback to KC with the code and state page showed an error, which should have been a GET. However, the KC was called back with other methods (POST or so) and showed the 405 error.

Since it's 405 and blocked from proceeding, the `authResponse` controller didn't have a chance to log anything.

**Comment by Jeff-Tian** [comment]
If you can access the corresponding keycloak IDP source code, then you can try add a `@Path("")` annotation to the `authResponse` method:

```diff
  @GET
+ @Path("")
  public Response authResponse( // ...
```

Reference: https://github.com/keycloak/keycloak/issues/25882#issuecomment-1892244104
